java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -309,14 +310,14 @@ private void visit(Tree tree, Tree terminator) {
         programState = ProgramState.unstack(programState, newArrayTree.initializers().size()).a;
         SymbolicValue svNewArray = constraintManager.createSymbolicValue(newArrayTree);
         programState = ProgramState.stackValue(programState, svNewArray);
-        programState = svNewArray.setConstraint(programState, NullConstraint.NOT_NULL);
+        programState = svNewArray.setSingleConstraint(programState, NullConstraint.NOT_NULL);
         break;
       case NEW_CLASS:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1151)